.github/workflows/build_and_push_api_image.yml

```diff
@@ -0,0 +1,73 @@
+name: "Build Moped API image and push to AWS ECR"
+on:
+  push:
+    # branches:
+    # - main
+    # - production
+    paths:
+      - ".github/workflows/build_and_push_api_image.yml"
+      - "api/**"
+  # Allows you to run this workflow manually from the Actions tab
+  workflow_dispatch:
+jobs:
+  build:
+    name: Build Moped API docker image
+    runs-on: ubuntu-latest
+    steps:
+      - name: Check out repository
+        uses: actions/checkout@v4
+
+      - name: Configure AWS credentials
+        uses: aws-actions/configure-aws-credentials@v4
+        with:
+          aws-access-key-id: ${{ secrets.AWS_ACCESS_KEY_ID }}
+          aws-secret-access-key: ${{ secrets.AWS_SECRET_ACCESS_KEY }}
+          aws-region: ${{ secrets.AWS_REGION || 'us-east-1' }}
+
+      - name: Login to Amazon ECR
+        id: login-ecr
+        uses: aws-actions/amazon-ecr-login@v2
+
+      - name: Determine image tag
+        id: determine-tag
+        run: |
+          if [ "${{ github.ref_name }}" = "production" ]; then
+            echo "tag=production" >> $GITHUB_OUTPUT
+          else
+            echo "tag=staging" >> $GITHUB_OUTPUT
+          fi
+
+      - name: Set up Docker Buildx
+        uses: docker/setup-buildx-action@v3
+
+      - name: Build and push image
+        uses: docker/build-push-action@v5
+        with:
+          platforms: linux/amd64,linux/arm64
+          push: true
+          tags: ${{ steps.login-ecr.outputs.registry }}/${{ secrets.ECR_REPOSITORY }}:${{ steps.determine-tag.outputs.tag }}
```

Comment:
Defining the non-default registry, `${{ steps.login-ecr.outputs.registry }}/`, is really the only meaningful change to make this work with ECR instead of docker hub.